File: for-local-test/342.power-of-four.java
class Solution {
    public boolean isPowerOfFour(int n) {
        if (n <= 0) return false;
        boolean isPowerOfTwo = (n & (n - 1)) == 0;
        return isPowerOfTwo && n % 3 == 1;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isPowerOfFour(16)); // true
        System.out.println(solution.isPowerOfFour(15)); // false
        System.out.println(solution.isPowerOfFour(1));  // true
        System.out.println(solution.isPowerOfFour(4));  // true
        System.out.println(solution.isPowerOfFour(8));  // false
    }
}

File: for-local-test/python/342.power-of-four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n <= 0:
            return False
        isPowerOfTwo = (n & (n - 1)) == 0
        return isPowerOfTwo and n % 3 == 1
    
if __name__ == "__main__":
    solution = Solution()
    print(solution.isPowerOfFour(16))  # True
    print(solution.isPowerOfFour(15))  # False
    print(solution.isPowerOfFour(1))   # True
    print(solution.isPowerOfFour(4))   # True
    print(solution.isPowerOfFour(8))   # False